src/debug.cc

```diff
@@ -62,6 +64,9 @@ class DebugFile
     /// logging stream; the only method that uses stderr as the last resort
     FILE *file() { return file_ ? file_ : stderr; }
 
+    /// whether a real log file was opened
+    bool opened() { return file_; }
```

Comment:
The name is ambiguous because a previously opened file may be closed now.

```suggestion
    bool storesMessages() const { return file_; }
```

or

```suggestion
    bool isOpen() const { return file_; }
```